source/core/src/main/com/csse3200/game/components/player/PlayerStatsDisplay.java

```diff
@@ -172,22 +198,49 @@ public void updatePlayerHealthUI(int health) {
     }
 
     public void updateMeleeWeaponUI() {
-        CharSequence text = String.format("Pickaxe x %d", 1);
-        pickaxeLabel.setText(text);
+        CharSequence text = String.format("Sword x %d", 1);
+        swordLabel.setText(text);
     }
 
-    public void updateRangedWeaponUI(int maxAmmo) {
-        CharSequence text = String.format("Shotgun x %d", 1);
+    /**
+     * Updates the ranged weapon UI.
+     * If the player is Player 4, show the pistol with 7 ammo.
+     *
+     * @param maxAmmo the maximum ammo count for the weapon.
+     * @param weaponName the name of the weapon (e.g., "Shotgun" or "Pistol").
+     */
+    public void updateRangedWeaponUI(int maxAmmo, String weaponName) {
```

Comment:
Again, player will only hold 1 range and 1 melee weapon at once, so this is strictly unnecessary